corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -54,16 +54,13 @@ hqDefine(
             $(".main-form").on("submit", function () {
                 var finalFormContent = getFormContents();
                 // compare changes with initialFormContent to check if only 'message' has changed
-                var formContentChanges = [];
-                Object.keys(finalFormContent).forEach(function (key) {
-                    if (finalFormContent[key] !== initialFormContent[key]) {
-                        formContentChanges.push(key);
+                var shouldInitiateRule = _.find(_.keys(finalFormContent), function (key) {
+                    if (finalFormContent[key] !== initialFormContent[key] && key !== "content-message") {
+                        return true;
                     }
                 });
-                if (!_.isEqual(["content-message"],formContentChanges)) {
-                    // to convey to HQ to reprocess or not
-                    $("#initiate_rule").val(false);
-                }
+                $("#initiate_rule").val(Boolean(shouldInitiateRule));
```

Comment:
why do we need the `Boolean` call here?

Reply:
Because we wanted the end result to be `true` or `false`. 
`_.find` returns the key for which the condition is true or it will return `undefined` if the condition does not match for any of the keys.

Reply:
cool, is that same as doing `!!shouldInitiateRule`?
(not saying you should change it to that, just understanding if this does something different)

Reply:
Yes. It's the same thing.